internal/engine/session.go

```diff
@@ -204,6 +208,11 @@ func NewSession(ctx context.Context, config SessionConfig) (*Session, error) {
 	return sess, nil
 }
 
+// TunnelDir returns the persistent directory used by tunnels.
+func (s *Session) TunnelDir() string {
+	return s.tunnelDir
+}
+
```

Comment:
I need to expose to experiments the directory used by persistent tunnels (e.g., the ones that up until this diff one could only leverage by using `miniooni --tunnel=TUNNEL ...`).

I am wondering whether we should create a lockfile here but I think `tor` probably already does that?

Reply:
What race condition are you concerned about? Multiple tor instances sharing the same data_dir?

Reply:
Yes, indeed! Now the data dir is `$OONI_HOME/tunnel/torsf` so two concurrent `torsf` tests would use it concurrently.

Reply:
So, I tried to run two instances of `tor` sharing the same directory. Here's what I've got:

```
Feb 07 12:36:38.000 [warn] It looks like another Tor process is running with the same data directory.  Waiting 5 seconds to see if it goes away.
Feb 07 12:36:43.000 [err] No, it's still there.  Exiting.
Feb 07 12:36:43.000 [err] set_options: Bug: Acting on config options left us in a broken state. Dying. (on Tor 0.4.6.9 )
Feb 07 12:36:43.000 [err] Reading config failed--see warnings above.
```

I think we're covered here. ✔️